docs/docs/aztec/concepts/accounts/keys.md

```diff
@@ -3,157 +3,125 @@ title: Keys
-:::
-
-:::note
-Additionally, there is typically a signing key pair which is used for authenticating the owner of the account.
-However, since Aztec supports native [account abstraction](../accounts/index.md#what-is-account-abstraction) this is not defined in protocol.
-Instead it's up to the account contract developer to implement it.
-:::
-
-## Public keys retrieval
-
-The keys for our accounts can be retrieved from the [Private eXecution Environment (PXE)](../pxe/index.md) using the following getter in Aztec.nr:
+### Nullifier keys 
 
-```
-fn get_public_keys(account: AztecAddress) -> PublicKeys;
-```
+Nullifier keys are presented as a pair of the master nullifier public key (`Npk_m`) and the master nullifier secret key (`nsk_m`). 
 
-It is necessary to first register the user as an account in our PXE, by calling the `registerAccount` PXE endpoint using Aztec.js, providing the account's secret key and partial address.
+To spend a note, the user computes a nullifier corresponding to this note. A nullifier is a hash of the note hash and app siloed nullifier secret key, the latter is derived using the nullifier master secret key.  To compute the nullifier, the protocol checks that the app siloed key is derived from the master key for this contract and that master nullifier public key is linked to the note owner's address. 
 
-During private function execution these keys are obtained via an oracle call from PXE.
+### Incoming viewing keys
 
-## Scoped keys
+The incoming viewing public key (`Ivpk`) is used by the sender to encrypt a note for the recipient. The corresponding incoming viewing secret key (`ivsk`) is used by the recipient to decrypt the note.
 
-To minimize damage of potential key leaks the keys are scoped (also called app-siloed) to the contract that requests them.
-This means that the keys used for the same user in two different application contracts will be different and potential leak of the scoped keys would only affect 1 application.
+When it comes to notes encryption and decryption:
+- For each note, there is a randomly generated ephemeral key pair (`esk`, `Epk`) where `Epk = esk * G`. 
+- The `AddressPublicKey` (derived from the `Ivsk`) together with `esk` are encrypted as a secret `S`, `S = esk * AddressPublicKey`.
+- `symmetric_encryption_key = hash(S)`
+- `Ciphertext = aes_encrypt(note, symmetric_encryption_key)`
+- The recipient gets a pair (`Epk`, `Ciphertext`)
+- The recipient uses the `address_sk` to decrypt the secret: `S = Epk * address_sk`.
+- The recipient uses the decrypted secret to decrypt the ciphertext.
 
-This also allows per-application auditability.
-A user may choose to disclose their incoming and outgoing viewing keys for a given application to an auditor or regulator (or for 3rd party interfaces, e.g. giving access to a block explorer to display my activity), as a means to reveal all their activity within that context, while retaining privacy across all other applications in the network.
+### Address keys
```

Comment:
It might be good to move this section above the section on 'Incoming view keys' since you explain where address_sk and AddressPublicKey come from in here, and you reference them above.